backend/agents/persona_agent.py

```
@@ -155,74 +161,142 @@ def get_scene_context(scene_description: str) -> str:
+                                if not doc.page_content.startswith("CONVERSATION_RESET_MARKER"):
+                                    history_parts.append(f"- {doc.page_content}")
+
+                    # Sort all documents by timestamp to ensure chronological order
+                    all_docs_with_timestamps = []
+                    for doc in all_docs:
+                        timestamp_str = doc.metadata.get('timestamp', '1970-01-01T00:00:00')
+                        try:
+                            from datetime import datetime
+                            timestamp = datetime.fromisoformat(timestamp_str.replace('Z', '+00:00'))
+                            all_docs_with_timestamps.append((timestamp, doc))
+                        except:
```

Ruff (0.14.1)

Do not use bare `except`

(E722)